core/tests/puppeteer-acceptance-tests/user-utilities/blog-post-admin-utils.js

```diff
@@ -0,0 +1,461 @@
+      for (let i = 0; i < allDraftBlogPosts.length; i++) {
+        let checkDraftBlogPostTitle = allDraftBlogPosts[i].
+          getElementsByClassName('e2e-test-blog-post-title')[0].innerText;
+        if (draftBlogPostTitle === checkDraftBlogPostTitle) {
+          allDraftBlogPosts[i].getElementsByClassName(
+            'e2e-test-blog-post-edit-box')[0].click();
+          await window.deleteDraftBlogPost();
+          return;
+        }
+      }
+    }, {draftBlogPostTitle});
+  }
+
+  /**
+   * This function checks if the Publish button is disabled.
+   */
+  async expectPublishButtonToBeDisabled() {
+    await this.page.waitForSelector('button.e2e-test-publish-blog-post-button');
+    await this.page.evaluate(() => {
+      const publishedButtonIsDisabled = document.getElementsByClassName(
+        'e2e-test-publish-blog-post-button')[0].disabled;
+      if (!publishedButtonIsDisabled) {
+        throw new Error(
+          'Published button is not disabled when the ' +
+          'blog post data is not completely filled');
+      }
+    });
+    showMessage(
+      'Published button is disabled when blog post data ' +
+      'is not completely filled.');
+  }
+
+  /**
+   * This function publishes a blog post with given title.
+   * @param {string} newBlogPostTitle - The title of the blog post
+   * to be published.
+   */
+  async publishNewBlogPostWithTitle(newBlogPostTitle) {
+    await this.addUserBioInBlogDashboard();
+    // See Note-1 below.
+    await this.page.waitForTimeout(500);
+    await this.clickOn(LABEL_FOR_NEW_BLOG_POST_CREATE_BUTTON);
+
+    await this.expectPublishButtonToBeDisabled();
+    await this.clickOn('button.mat-button-toggle-button');
+    await this.expectPublishButtonToBeDisabled();
+    await this.clickOn(thumbnailPhotoBox);
+    await this.uploadFile('../images/blog-post-thumbnail.svg');
+    await this.clickOn(LABEL_FOR_ADD_THUMBNAIL_BUTTON);
+    await this.page.waitForSelector('body.modal-open', {hidden: true});
+    await this.expectPublishButtonToBeDisabled();
+
+    await this.type(blogTitleInput, newBlogPostTitle);
+    await this.page.keyboard.press('Tab');
+    await this.type(blogBodyInput, 'test blog post body content');
+    await this.clickOn(LABEL_FOR_DONE_BUTTON);
+
+    await this.page.waitForSelector(
+      'button.e2e-test-publish-blog-post-button:not([disabled])');
+    await this.clickOn('PUBLISH');
+    await this.page.waitForSelector('button.e2e-test-confirm-button');
+    await this.clickOn(LABEL_FOR_CONFIRM_BUTTON);
+    showMessage('Successfully published a blog post!');
+  }
+
+  /**
+   * This function deletes a published blog post with the given title.
+   * @param {string} blogPostTitle - The title of the published blog post
+   * to be deleted.
+   */
+  async deletePublishedBlogPostWithTitle(blogPostTitle) {
+    await this.clickOn('PUBLISHED');
+    await this.page.exposeFunction('deletePublishedBlogPost', async() => {
+      // See Note-2 below.
```

Comment:
Ditto here and elsewhere. If you do something weird in the tests then each case must be justified carefully. Where possible, avoid doing something weird, and try to wait for a condition to happen rather than a fixed timeout.

Reply:
done.